running pipeline-in-pipeline controller in another namespace doesn't work

I tried running the yaml in another namespace, simply by doing this:

```
sed -i 's/namespace: tekton-pip-run/namespace: somens/g' /tmp/pip.yaml
```

And I'm getting this error:
<details>

<summary>output</summary>


```bash
2021/09/02 17:46:41 Registering 3 clients
2021/09/02 17:46:41 Registering 2 informer factories
2021/09/02 17:46:41 Registering 2 informers
2021/09/02 17:46:41 Registering 1 controllers
{"level":"info","ts":1630604801.7829762,"caller":"logging/config.go:115","msg":"Successfully created the logger.","logging.googleapis.com/labels":{},"logging.googleapis.com/sourceLocation":{"file":"knative.dev/pkg@v0.0.0-20210114223020-f0ea5e6b9c4e/logging/config.go","line":"115","function":"knative.dev/pkg/logging.newLoggerFromConfig"}}
{"level":"info","ts":1630604801.7830176,"caller":"logging/config.go:116","msg":"Logging level set to: info","logging.googleapis.com/labels":{},"logging.googleapis.com/sourceLocation":{"file":"knative.dev/pkg@v0.0.0-20210114223020-f0ea5e6b9c4e/logging/config.go","line":"116","function":"knative.dev/pkg/logging.newLoggerFromConfig"}}
{"level":"info","ts":1630604801.783083,"caller":"logging/config.go:78","msg":"Fetch GitHub commit ID from kodata failed","error":"open /var/run/ko/HEAD: no such file or directory","logging.googleapis.com/labels":{},"logging.googleapis.com/sourceLocation":{"file":"knative.dev/pkg@v0.0.0-20210114223020-f0ea5e6b9c4e/logging/config.go","line":"78","function":"knative.dev/pkg/logging.enrichLoggerWithCommitID"}}
{"level":"info","ts":1630604801.78312,"logger":"pip-controller","caller":"profiling/server.go:64","msg":"Profiling enabled: false","logging.googleapis.com/labels":{},"logging.googleapis.com/sourceLocation":{"file":"knative.dev/pkg@v0.0.0-20210114223020-f0ea5e6b9c4e/profiling/server.go","line":"64","function":"knative.dev/pkg/profiling.NewHandler"}}
{"level":"info","ts":1630604801.7862043,"logger":"pip-controller","caller":"leaderelection/context.go:46","msg":"Running with Standard leader election","logging.googleapis.com/labels":{},"logging.googleapis.com/sourceLocation":{"file":"knative.dev/pkg@v0.0.0-20210114223020-f0ea5e6b9c4e/leaderelection/context.go","line":"46","function":"knative.dev/pkg/leaderelection.WithDynamicLeaderElectorBuilder"}}
{"level":"info","ts":1630604801.7863595,"logger":"pip-controller","caller":"pip/controller.go:59","msg":"Setting up event handlers","logging.googleapis.com/labels":{},"logging.googleapis.com/sourceLocation":{"file":"github.com/tektoncd/experimental/pipelines-in-pipelines/pkg/reconciler/pip/controller.go","line":"59","function":"github.com/tektoncd/experimental/pipelines-in-pipelines/pkg/reconciler/pip.NewController"}}
panic: The environment variable "METRICS_DOMAIN" is not set
If this is a process running on Kubernetes, then it should be specifying
this via:
env:
- name: METRICS_DOMAIN
value: knative.dev/some-repository
If this is a Go unit test consuming metric.Domain() then it should add the
following import:
import (
_ "knative.dev/pkg/metrics/testing"
)
goroutine 1 [running]:
knative.dev/pkg/metrics.Domain(0xc00094f7c0, 0x1892e0d)
knative.dev/pkg@v0.0.0-20210114223020-f0ea5e6b9c4e/metrics/config.go:291 +0xff
knative.dev/pkg/metrics.ConfigMapWatcher(0x1f9e298, 0xc000172ae0, 0x1cc0953, 0xe, 0xc00000c3a8, 0xc000186708, 0x0)
knative.dev/pkg@v0.0.0-20210114223020-f0ea5e6b9c4e/metrics/exporter.go:99 +0x26
knative.dev/pkg/injection/sharedmain.WatchObservabilityConfigOrDie(0x1f9e298, 0xc000172ae0, 0xc00057ff80, 0xc000a63d40, 0xc000186708, 0x1cc0953, 0xe)
knative.dev/pkg@v0.0.0-20210114223020-f0ea5e6b9c4e/injection/sharedmain/main.go:328 +0x546
knative.dev/pkg/injection/sharedmain.MainWithConfig(0x1f9e298, 0xc000172ae0, 0x1cc0953, 0xe, 0xc0001d6000, 0xc00094ff70, 0x1, 0x1)
knative.dev/pkg@v0.0.0-20210114223020-f0ea5e6b9c4e/injection/sharedmain/main.go:201 +0x706
knative.dev/pkg/injection/sharedmain.MainWithContext(0x1f9ed88, 0xc0001c4000, 0x1cc0953, 0xe, 0xc00094ff70, 0x1, 0x1)
knative.dev/pkg@v0.0.0-20210114223020-f0ea5e6b9c4e/injection/sharedmain/main.go:142 +0xda
knative.dev/pkg/injection/sharedmain.Main(0x1cc0953, 0xe, 0xc00094ff70, 0x1, 0x1)
knative.dev/pkg@v0.0.0-20210114223020-f0ea5e6b9c4e/injection/sharedmain/main.go:116 +0x9c
main.main()
github.com/tektoncd/experimental/pipelines-in-pipelines/cmd/controller/main.go:25 +0x67
```

</details>

I used the file from here: https://storage.cloud.google.com/tekton-releases-nightly/pipelines-in-pipelines/latest/release.yaml


I don't see the env var: `METRICS_DOMAIN` anywhere and I don't know why it's needed, this only happens if I change the namespace, otherwise it works perfectly.

--- a/cel/config/500-controller.yaml
+++ b/cel/config/500-controller.yaml
@@ -60,6 +60,8 @@ spec:
             # values in the "configmaps" "get" rule.
             - name: CONFIG_LOGGING_NAME
               value: config-logging
+            - name: METRICS_DOMAIN
+              value: experimental.tekton.dev/cel
       volumes:
         - name: config-logging
           configMap:

--- a/cloudevents/config/controller.yaml
+++ b/cloudevents/config/controller.yaml
@@ -74,6 +74,8 @@ spec:
         # values in the "configmaps" "get" rule.
         - name: CONFIG_LOGGING_NAME
           value: config-logging
+        - name: METRICS_DOMAIN
+          value: experimental.tekton.dev/cloudevents
       volumes:
         - name: config-logging
           configMap:

--- a/pipeline-to-taskrun/config/500-controller.yaml
+++ b/pipeline-to-taskrun/config/500-controller.yaml
@@ -60,6 +60,8 @@ spec:
             # values in the "configmaps" "get" rule.
             - name: CONFIG_LOGGING_NAME
               value: config-logging
+            - name: METRICS_DOMAIN
+              value: experimental.tekton.dev/pipeline-to-taskrun
       volumes:
         - name: config-logging
           configMap:

--- a/pipelines-in-pipelines/config/500-controller.yaml
+++ b/pipelines-in-pipelines/config/500-controller.yaml
@@ -60,6 +60,8 @@ spec:
             # values in the "configmaps" "get" rule.
             - name: CONFIG_LOGGING_NAME
               value: config-logging
+            - name: METRICS_DOMAIN
+              value: experimental.tekton.dev/pipelines-in-pipelines
       volumes:
         - name: config-logging
           configMap:

--- a/task-loops/config/500-controller.yaml
+++ b/task-loops/config/500-controller.yaml
@@ -67,8 +67,7 @@ spec:
         - name: CONFIG_OBSERVABILITY_NAME
           value: config-observability
         - name: METRICS_DOMAIN
-          value: tekton.dev/pipeline
+          value: experimental.tekton.dev/task-loops
         securityContext:
           allowPrivilegeEscalation: false
           runAsUser: 1001
-

--- a/task-loops/config/500-webhook.yaml
+++ b/task-loops/config/500-webhook.yaml
@@ -70,7 +70,7 @@ spec:
         - name: WEBHOOK_SECRET_NAME
           value: tekton-taskloop-webhook-certs
         - name: METRICS_DOMAIN
-          value: tekton.dev/pipeline
+          value: experimental.tekton.dev/task-loops
         securityContext:
           allowPrivilegeEscalation: false
           runAsUser: 1001

--- a/wait-task/config/controller.yaml
+++ b/wait-task/config/controller.yaml
@@ -202,6 +202,8 @@ spec:
         # values in the "configmaps" "get" rule.
         - name: CONFIG_LOGGING_NAME
           value: config-logging
+        - name: METRICS_DOMAIN
+          value: experimental.tekton.dev/wait-task
       volumes:
         - name: config-logging
           configMap:

